src/number_theory/wilson.lean

```diff
@@ -0,0 +1,48 @@
+/-
+Copyright (c) 2022 John Nicol. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: John Nicol
+-/
+import number_theory.legendre_symbol.gauss_eisenstein_lemmas
+
+/-!
+# Wilson's theorem.
+
+This file contains a proof of Wilson's theorem.
+
+The heavy lifting is mostly done by the previous `wilsons_lemma`,
+but here we also prove the other logical direction.
+
+This could be generalized to similar results about finite abelian groups.
+
```

Comment:
> Is it possible to refactor the older code separately? I think this is a straightforward, standalone commit.

I think it would be fine to do a refactor PR after this so long as you add a `## Todo` section naming which lemma should move into this module.